### PR TITLE
Disable add new item from Dictionary Editor when there is an invalid key.

### PR DESCRIPTION
**Pull request by ajreckof**
Small step towards #76078
Commentary stating the need of this fix : https://github.com/godotengine/godot/pull/88636#pullrequestreview-1894210868

I'm not confident at all but the message in the tooltip so feel to give ideas for a better tootltip. The one thing I feel is missing is that I don't know at all why tis behaviour is even needed :(. I tracked the addition to #20400 but there is no reason given.

### CODE REVIEW

File: editor/editor_properties_array_dict.h [hunk 1]
```diff
@@ -72,6 +73,10 @@ class EditorPropertyDictionaryObject : public RefCounted {
 	void set_new_item_key(const Variant &p_new_item);
 	Variant get_new_item_key();
 
+	void update_new_item_key_invalidity();
+	bool get_new_item_key_invalidity();
+	Callable validity_change_callback;
```
Comment (AThousandShips) [inline editor/editor_properties_array_dict.h:78]: Should this really be public?

Comment (ajreckof) [reply editor/editor_properties_array_dict.h:78]: I don't how it would work without it being public since it needs the EditorProperty to set the callback.

Comment (AThousandShips) [reply editor/editor_properties_array_dict.h:78]: That sounds like something that should use a method, exposing variables directly is not the general appropriate way to handle things in c++

**Comment by ajreckof** [comment]
I'm not sure this is the way to go as I don't see a good reasons we are preventing it.

**Pull request closed without merging**